server/src/main/java/org/apache/cassandra/sidecar/db/SidecarPermissionsDatabaseAccessor.java

```diff
@@ -0,0 +1,92 @@
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.db;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.datastax.driver.core.BoundStatement;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import io.vertx.ext.auth.authorization.Authorization;
+import org.apache.cassandra.sidecar.common.server.CQLSessionProvider;
+import org.apache.cassandra.sidecar.db.schema.SidecarRolePermissionsSchema;
+
+import static org.apache.cassandra.sidecar.utils.AuthUtils.actionFromName;
+
+/**
+ * {@link SidecarPermissionsDatabaseAccessor} is an accessor for role_permissions_v1 table under sidecar_internal
+ * keyspace. Custom sidecar specific permissions are stored in this table.
+ */
+@Singleton
+public class SidecarPermissionsDatabaseAccessor extends DatabaseAccessor<SidecarRolePermissionsSchema>
+{
+    private static final Logger logger = LoggerFactory.getLogger(SidecarPermissionsDatabaseAccessor.class);
+
+    @Inject
+    protected SidecarPermissionsDatabaseAccessor(SidecarRolePermissionsSchema tableSchema,
+                                                 CQLSessionProvider sessionProvider)
+    {
+        super(tableSchema, sessionProvider);
+    }
+
+    /**
+     * Queries Sidecar for all rows in sidecar_internal.role_permissions_v1 table. This table maps permissions of a
+     * role into {@link Authorization} and returns a {@code Map} of user role to authorizations.
+     *
+     * @return - {@code Map} contains role and granted authorizations
+     */
+    public Map<String, Set<Authorization>> getAllRolesAndPermissions()
+    {
+        BoundStatement statement = tableSchema.getAllRolesAndPermissions().bind();
+        ResultSet result = execute(statement);
+        Map<String, Set<Authorization>> roleAuthorizations = new HashMap<>();
+        for (Row row : result)
+        {
+            String role = row.getString("role");
+            String resource = row.getString("resource");
+            Set<String> permissions = row.getSet("permissions", String.class);
+            Set<Authorization> authorizations = new HashSet<>();
+            for (String permission : permissions)
+            {
+                try
+                {
+                    authorizations.add(actionFromName(permission).toAuthorization(resource));
+                }
+                catch (Exception e)
+                {
+                    logger.error("Error reading sidecar permission {} for resource {} for role {}, e",
+                                 permission, resource, role, e);
+                }
+            }
+            if (!authorizations.isEmpty())
+            {
+                roleAuthorizations.computeIfAbsent(role, k -> new HashSet<>(authorizations)).addAll(authorizations);
```

Comment:
If it is the first time role is getting added, then `addAll` will be called implicitly from the construstor and same set of elements will bee added again with `.addAll` outside. How about this?
```suggestion
                roleAuthorizations.computeIfAbsent(role, k -> new HashSet<>(authorizations.size())).addAll(authorizations);
```